core/src/main/java/com/opensymphony/xwork2/DefaultActionProxy.java

```diff
@@ -145,15 +145,15 @@ public String getNamespace() {
 
     public String execute() throws Exception {
         ActionContext nestedContext = ActionContext.getContext();
-        ActionContext.setContext(invocation.getInvocationContext());
+        ActionContext.bound(invocation.getInvocationContext());
```

Comment:
With using `bound` I want to better express that _the passed context will be bound to the current thread_ - this is what's happening when you call `setContext()` and this is also the main issue now, because in some parts of the code a new context can be created but it won't be _bound_ to the thread. I would like have this flow more simplified, but rather not in this PR.

Reply:
And having return value allows to _chain_ methods on the same instance - this is not used yet but I feel I will need that in the next iteration :)